src/NuGet.Clients/NuGet.VisualStudio.Common/OutputConsoleLogger.cs

```diff
@@ -139,27 +170,33 @@ private async Task<int> GetMSBuildVerbosityLevelAsync()
             return DefaultVerbosityLevel;
         }
 
-        public void Start()
+        public void ReportError(string message)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
+            Run($"{nameof(ReportError)}/{nameof(String)}", async () =>
             {
-                await OutputConsole.ActivateAsync();
-                await OutputConsole.ClearAsync();
-                _verbosityLevel = await GetMSBuildVerbosityLevelAsync();
-                ErrorListTableDataSource.Value.ClearNuGetEntries();
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                var errorListEntry = new ErrorListTableEntry(message, LogLevel.Error);
+                ErrorListTableDataSource.Value.AddNuGetEntries(errorListEntry);
             });
         }
 
-        public void ReportError(string message)
+        public void ReportError(ILogMessage message)
         {
-            var errorListEntry = new ErrorListTableEntry(message, LogLevel.Error);
-            ErrorListTableDataSource.Value.AddNuGetEntries(errorListEntry);
+            Run($"{nameof(ReportError)}/{nameof(ILogMessage)}", async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                var errorListEntry = new ErrorListTableEntry(message);
+                ErrorListTableDataSource.Value.AddNuGetEntries(errorListEntry);
+            });
         }
 
-        public void ReportError(ILogMessage message)
+        private void Run(string methodName, Func<Task> action)
```

Comment:
tip: Write this signature as:

```cs
Func<Task> action, [CallerMemberName] string methodName = null
```

Then you can stop actually specifying the method name in every caller that only supplied the method name because the C# compiler will automatically supply it.